assembly/src/ast/nodes/advice.rs

```diff
@@ -152,6 +157,9 @@ impl Deserializable for AdviceInjectorNode {
                 let domain = source.read_u8()?;
                 Ok(AdviceInjectorNode::InsertHdwordImm { domain })
             }
+            PUSH_SIGN => Ok(AdviceInjectorNode::PushSignature {
+                kind: SignatureKind::RpoFalcon512,
+            }),
```

Comment:
It might be good to write the actual serialization and deserialization code for `SignatureKind`. The pattern we can use could be something similar to what I used for serializing/deserializing `DebugOptions` [here](https://github.com/0xPolygonMiden/miden-vm/blob/fc7b247e28b640ea6206c86efc9044e2041f4f25/assembly/src/ast/nodes/serde/debug.rs).

I know we have just a single signature scheme for now, but it would make it more "future-compatible".

Reply:
Where should this be placed?